### PR TITLE
Migrated to using snakbar with ScaffoldMessenger in provider_shopper

### PR DESCRIPTION
**Pull request by Adityabhaumik**
I also changed the color of the buy button  white -> black as it was not visible properly.
issue : Update provider_shopper for upcoming release #657

### CODE REVIEW

**Comment by RedBrogdon** [comment]
@Adityabhaumik, this change won't work for stable, which means it should land in our beta branch for the time being. Can you change the base branch to `beta`?

See this link for details on how to do it in GitHub's UI: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request

**Comment by Adityabhaumik** [comment]
@RedBrogdon I changed the base branch to beta.

File: provider_shopper/lib/screens/cart.dart [hunk 1]
```diff
@@ -85,11 +85,18 @@ class _CartTotal extends StatelessWidget {
             SizedBox(width: 24),
             TextButton(
               onPressed: () {
-                Scaffold.of(context).showSnackBar(
-                    SnackBar(content: Text('Buying not supported yet.')));
+                //Updated Method of using Snack Bar for more info visit "https://flutter.dev/docs/release/breaking-changes/scaffold-messenger"
```
Comment (RedBrogdon) [inline provider_shopper/lib/screens/cart.dart:88]: You can leave this comment out. If we added links every time the API changed, we'd have run out of room for the actual code by now. 😄

File: provider_shopper/lib/screens/cart.dart [hunk 2]
```diff
@@ -85,11 +85,18 @@ class _CartTotal extends StatelessWidget {
...
+                  duration: const Duration(seconds: 1),
+                  action: SnackBarAction(
+                    label: ' ',
+                    onPressed: () { },
+                  ),
+                ));
               },
               style: TextButton.styleFrom(primary: Colors.white),
-              child: Text('BUY'),
+              child: Text('BUY',style: TextStyle(fontWeight: FontWeight.bold,color: Colors.black),),
```
Comment (RedBrogdon) [inline provider_shopper/lib/screens/cart.dart:99]: This needs to be reformatted. Add a comma after `Colors.black` and then run `flutter format` in your IDE or on the command line.

**Review by RedBrogdon — CHANGES_REQUESTED** [review]

**Comment by RedBrogdon** [comment]
Thanks for swapping the branch, @Adityabhaumik. I added a couple minors comments, but this should be ready once those are resolved.

**Comment by Adityabhaumik** [comment]
@RedBrogdon I made those changes but I don't understand why there was a merge conflict. 😥

File: provider_shopper/lib/screens/cart.dart [hunk 3]
```diff
@@ -85,11 +85,24 @@ class _CartTotal extends StatelessWidget {
             SizedBox(width: 24),
             TextButton(
               onPressed: () {
-                ScaffoldMessenger.of(context).showSnackBar(
-                    SnackBar(content: Text('Buying not supported yet.')));
+                  ScaffoldMessenger.of(context).showSnackBar(SnackBar(
```
Comment (RedBrogdon) [inline provider_shopper/lib/screens/cart.dart:88]: This code doesn't look like it's formatted correctly. Can you run `flutter format` from the command line or trigger a reformat in your IDE?

**Review by RedBrogdon — CHANGES_REQUESTED** [review]

**Comment by RedBrogdon** [comment]
> @RedBrogdon I made those changes but I don't understand why there was a merge conflict. 😥

I'm not sure either. It doesn't look like there's a merge conflict anymore, though.

File: provider_shopper/lib/screens/cart.dart (hunk 3 above)
Comment (Adityabhaumik) [reply provider_shopper/lib/screens/cart.dart]: this was the code I deleted to  resolve the merge request directly from the Github website that is why it is not formatted I guess

**Comment by Adityabhaumik** [comment]
I think I messed up with the previous commit and I can't figure out why there are so many changes, even though I only pushed the cart page 😥.  Sorry! I am very new to open source

**Comment by RedBrogdon** [comment]
> Sorry! I am very new to open source

That's okay. We were all new at one point. 😄 

It looks like a bunch of extra commits have ended up in this PR, probably due to the fact that the beta branch was updated and force-pushed while this PR was already staged. Because the changes you're trying to land are relatively simple, the easiest thing to do is just close this PR and create another one with the same changes:

1) I'll close this PR.
2) You can make sure you have an up-to-date version of the beta branch on your local machine.
3) You can make a new branch based off of beta.
4) Add your changes to that branch.
5) Create a new PR with those changes, and have it target the beta branch.

That process should get you a simple, clean PR with just the changes you want.

**Pull request closed without merging**